cypress/integration/plugins/anomaly-detection-dashboards-plugin/create_detector_spec.js

```diff
@@ -80,14 +80,10 @@ context('Create detector workflow', () => {
     cy.getElementByTestId('featureTable').contains(TEST_FEATURE_NAME);
     cy.getElementByTestId('createDetectorButton').click();
 
-    // Clean up the created detector. Extract detector ID from the detector configuration page
-    cy.getElementByTestId('detectorIdCell').within(() => {
-      cy.get('.euiText--medium')
-        .invoke('text')
-        .then((detectorId) => {
-          cy.log('Deleting detector with ID: ' + detectorId);
-          cy.deleteDetector(detectorId);
-        });
-    });
+    // Lands on the config page by default. Delete the detector to clean up.
+    cy.getElementByTestId('actionsButton').click();
+    cy.getElementByTestId('deleteDetectorItem').click();
```

Comment:
Will it wait for delete detector button to show up or basically waits for it to load the config page before trying these clicks

Reply:
It should retry by default.

Reply:
The page and all of the components, including the dropdown with the delete option, will need to load and be functional before it can execute all of this. In other words, if following this sequence of events fails to delete the detector, then that is a separate UI bug that would need to be addressed, and wouldn't be an issue with race conditions in the test